apps/studio/src/server/modules/resource/resource.service.ts

```diff
@@ -84,6 +85,7 @@ export const getFullPageById = async (args: {
     .where("Resource.mainBlobId", "is not", null)
     .innerJoin("Blob", "Resource.mainBlobId", "Blob.id")
     .select(defaultResourceWithBlobSelect)
+    .forUpdate()
```

Comment:
under what context are we using forUpdate? Isn't this to be used within a transaction? I don't think this is a transaction right?

Reply:
I actually am not sure if `forUpdate` is meant to be used in a tx. 

Trying it on local works but if what you say is correct, then I think the only choice here would be to write a wrapper method with `forUpdate` in a tx

Reply:
Kysely docs don't provide info on this too

Reply:
Hmm locking mechanics happen within a transaction right

> FOR UPDATE causes the rows retrieved by the SELECT statement to be locked as though for update. This prevents them from being locked, modified or deleted by other transactions until the current transaction ends. 

Mentions "current transaction ends"

@karrui can advise here?

Reply:
So usually with SQL we do smth like 

```
BEGIN tx

Select clause with forUpdate

do the update

COMMIT tx
```

Reply:
actually lemme double check w/ kr on this - i'll update you later

Reply:
yes. just follow postgres docs. forUpdate is specifically for a transaction. It locks any rows queries until the transaction ends, which is what we want to prevent other readers from accessing the row, preventing skew.